pkg/image/apis/image/validation/validation.go

```diff
@@ -153,13 +157,24 @@ func ValidateImageSignatureUpdate(newImageSignature, oldImageSignature *imageapi
 
 // ValidateImageStream tests required fields for an ImageStream.
 func ValidateImageStream(stream *imageapi.ImageStream) field.ErrorList {
+	return ValidateImageStreamWithWhitelister(nil, stream)
+}
+
+// ValidateImageStreamWithWhitelister tests required fields for an ImageStream. Additionally, it validates
+// each new image reference against registry whitelist.
+func ValidateImageStreamWithWhitelister(
+	whitelister whitelist.RegistryWhitelister,
+	stream *imageapi.ImageStream,
+) field.ErrorList {
 	result := validation.ValidateObjectMeta(&stream.ObjectMeta, true, ValidateImageStreamName, field.NewPath("metadata"))
 
 	// Ensure we can generate a valid docker image repository from namespace/name
 	if len(stream.Namespace+"/"+stream.Name) > reference.NameTotalLengthMax {
 		result = append(result, field.Invalid(field.NewPath("metadata", "name"), stream.Name, fmt.Sprintf("'namespace/name' cannot be longer than %d characters", reference.NameTotalLengthMax)))
 	}
 
+	insecureRepository := isRepositoryInsecure(stream)
+
 	if len(stream.Spec.DockerImageRepository) != 0 {
```

Comment:
we don't whitelist check stream.Spec.DockerImageRepository itself?  Does something else prevent a user from importing a disallowed image via that mechanism?

Reply:
There's a prevention mechanism - the individual tags get refused. But I agree it would be better to admit this spec as well to avoid needless frequent import errors. I'll fix this.